alchemy/src/aws/bucket.ts

```
@@ -127,108 +114,89 @@ export interface Bucket extends Resource<"s3::Bucket">, BucketProps {
+    } else {
+      // Create bucket if it doesn't exist
+      yield* client.put(`/${props.bucketName}`);
+
+      // Add tags after creation if specified
+      if (props.tags) {
+        const taggingXml = createTaggingXml(props.tags);
+        yield* client.put(`/${props.bucketName}?tagging`, taggingXml, {
+          "Content-Type": "application/xml",
+        });
       }
     }
 
-    // Get bucket details
+    // Get bucket details in parallel
     const [locationResponse, versioningResponse, aclResponse] =
-      await Promise.all([
-        retry(() =>
-          client.send(
-            new GetBucketLocationCommand({ Bucket: props.bucketName }),
-          ),
-        ),
-        retry(() =>
-          client.send(
-            new GetBucketVersioningCommand({ Bucket: props.bucketName }),
-          ),
-        ),
-        retry(() =>
-          client.send(new GetBucketAclCommand({ Bucket: props.bucketName })),
-        ),
+      yield* Effect.all([
+        client.get(`/${props.bucketName}?location`),
+        client.get(`/${props.bucketName}?versioning`),
+        client.get(`/${props.bucketName}?acl`),
       ]);
 
-    const region = locationResponse.LocationConstraint || "us-east-1";
+    const region = (locationResponse as any)?.LocationConstraint || "us-east-1";
 
-    // Get tags if they exist
+    // Get tags if they weren't provided
     let tags = props.tags;
     if (!tags) {
-      try {
-        const taggingResponse = await retry(() =>
-          client.send(
-            new GetBucketTaggingCommand({ Bucket: props.bucketName }),
-          ),
-        );
-        tags = Object.fromEntries(
-          taggingResponse.TagSet?.map(({ Key, Value }) => [Key, Value]) || [],
-        );
-      } catch (error: any) {
-        if (error.name !== "NoSuchTagSet") {
-          throw error;
+      const taggingResponse = yield* client
+        .get(`/${props.bucketName}?tagging`)
+        .pipe(Effect.catchAll(() => Effect.succeed(null)));
```

Why are we swallowing errors? We need to focus on how we handle errors